devito/types/dense.py

```diff
@@ -920,21 +930,24 @@ class Function(DiscreteFunction, Differentiable):
 
     is_Function = True
 
-    def __init__(self, *args, **kwargs):
-        if not self._cached():
-            super(Function, self).__init__(*args, **kwargs)
+    def _cache_meta(self):
+        # Attach additional metadata to self's cache entry
+        return {'nbytes': self.size}
 
-            # Space order
-            space_order = kwargs.get('space_order', 1)
-            if isinstance(space_order, int):
-                self._space_order = space_order
-            elif isinstance(space_order, tuple) and len(space_order) == 3:
-                self._space_order, _, _ = space_order
-            else:
-                raise TypeError("`space_order` must be int or 3-tuple of ints")
+    def __init_finalize__(self, *args, **kwargs):
```

Comment:
note for reviewers: this is exactly like before, but de-indented as we don't need ``if not self._cached()`` anymore